src/Microsoft.AspNetCore.Http.Connections/HttpConnectionDispatcher.cs

```diff
@@ -487,6 +492,36 @@ private static bool ServerHasWebSockets(IFeatureCollection features)
             }
         }
 
+        private async Task ProcessDeleteAsync(HttpContext context)
+        {
+            var connection = await GetConnectionAsync(context);
+            if (connection == null)
+            {
+                // No such connection, GetConnection already set the response status code
+                return;
+            }
+
+            // This end point only works for long polling
+            if (connection.TransportType != HttpTransportType.LongPolling)
+            {
+                Log.ReceivedDeleteRequestForUnsupportedTransport(_logger, connection.TransportType);
+                context.Response.StatusCode = StatusCodes.Status400BadRequest;
+                context.Response.ContentType = "text/plain";
+                await context.Response.WriteAsync("Cannot terminate this connection using the DELETE endpoint.");
+                return;
+            }
+
+            Log.TerminatingConection(_logger);
+
+            // Complete the receiving end of the pipe
+            connection.Application.Output.Complete();
+
+            // Dispose the connection gracefully, but don't wait for it
+            _ = connection.DisposeAsync(closeGracefully: true);
```

Comment:
I think this needs to be call `_ =_manager.DisposeAndRemoveAsync(connection, closeGracefully: false);`, otherwise nobody will remove the disposed connection from the list.